src/cppcache/src/Struct.cpp

```diff
@@ -96,16 +99,16 @@ Serializable* Struct::fromData(DataInput& input) {
   return this;
 }
 
-const char* Struct::getFieldName(const int32_t index) const {
+const std::string& Struct::getFieldName(const int32_t index) const {
   if (m_parent != nullptr) {
     return m_parent->getFieldName(index);
   } else {
     for (const auto& iter : m_fieldNames) {
-      if ((iter.second)->value() == index) return iter.first->asChar();
+      if (iter.second == index) return (iter.first);
     }
   }
 
-  return nullptr;
+  return std::string("");
```

Comment:
What about throwing [`std::out_of_range`](http://en.cppreference.com/w/cpp/error/out_of_range). I don't like the idea of checking for empty string.